src/react/tests/useQuery.test.tsx

```diff
@@ -737,6 +737,41 @@ describe('useQuery', () => {
     expect(states[1].dataUpdatedAt).not.toBe(states[2].dataUpdatedAt)
   })
 
+  it('should track properties and only re-render when a tracked property changes', async () => {
```

Comment:
Might be good to add some additional tests to see if the referential integrity stays the same and to see how it interacts with the other `notifyOnChange*` props. And we also need to define and test the behavior: will the hook remember each accessed property during its lifespan? Or will it reset on each render?

Reply:
> add some additional tests

will definitely do that.

> will the hook remember each accessed property during its lifespan?

I think it needs to reset and start new tracking with every render. it might be minor, but consider the following example:

```
const queryInfo = useQuery(...)

if (queryInfo.isLoading) return "Loading...";

if (queryInfo.isError)
  return "An error has occurred: " + queryInfo.error.message + queryInfo.dataUpdatedAt;

return <div>query.data ...</div>

```

in the first render, we would only track `isLoading` because of the early return. if we then get an error, we would track `isLoading, isError, error, dataUpdatedAt`. and if we then get data on the next refetch, we would want to track `isLoading, isError, data`, but if we don't reset, we would also keep tracking `error, dataUpdatedAt` properties, which doesn't seem right since we are not currently observing them

Reply:
> Might be good to add some additional tests to see if the referential integrity stays the same

fe0b4dc7

Reply:
here is another test for a combination of tracking props and excluding props: exclusion takes precedence, which I think is good: 5d923ce

so even is you track props and use one, if you exclude it explicitly, it will not re-render.